src/CFamily/CMake/CMakeHashCalculator.cs

```diff
@@ -0,0 +1,71 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace SonarLint.VisualStudio.CFamily.CMake
+{
+    internal static class CMakeHashCalculator
+    {
+        /// <summary>
+        /// Generates a GUID from the supplied input
+        /// </summary>
+        /// <remarks>The algorithm is equivalent to the one used by VS2019/2022 when calculating the
+        /// CMake macros "workspaceHash" and "projectHash".
+        /// <para>
+        /// The VS2019 implementation is in:
+        /// * assembly: \Common7\IDE\CommonExtensions\Microsoft\EBF\Microsoft.VisualStudio.Workspace.ExternalBuildFramework.Impl.dll
+        /// * method: HashHelper.GetGuid
+        /// Unfortunately, it isn't currently available in a NuGet package.
+        /// </para>
+        /// </remarks>
+        public static Guid CalculateVS2019Guid(string input)
```

Comment:
There isn't a lot to say about the algorithm - it does logically same as the VS version but the implementation is ours to comply with our corporate policies.